membership-attribute-service/app/services/MobileSubscriptionService.scala

```diff
@@ -0,0 +1,42 @@
+package services
+
+import configuration.Config
+import models.MobileSubscriptionStatus
+import com.github.nscala_time.time.OrderingImplicits._
+import play.api.libs.json.{JsError, JsSuccess}
+import play.api.libs.ws.WSClient
+import scalaz.{-\/, \/, \/-}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait MobileSubscriptionService {
+
+  def getSubscriptionStatusForUser(identityId: String): Future[\/[String, Option[MobileSubscriptionStatus]]]
+
+}
+
+class MobileSubscriptionServiceImpl(wsClient: WSClient)(implicit ec: ExecutionContext) extends MobileSubscriptionService {
+
+  override def getSubscriptionStatusForUser(identityId: String): Future[String \/ Option[MobileSubscriptionStatus]] = {
+    val response = wsClient.url(s"${Config.Mobile.subscriptionURL}/user/subscriptions/$identityId")
+      .withHttpHeaders("Authorization" -> s"Bearer ${Config.Mobile.subscriptionApiKey}")
+      .get()
+
+    response.map { resp =>
+      if (resp.status != 200) {
+        \/.left(s"Unable to fetch the mobile subscription status for $identityId")
```

Comment:
would be useful to log the body so we know what was return by mobile api

Reply:
some http libraries actually keep the inputstream lying around if you don't read it (okhttp) so it might be worth checking that